includes/class-fs-api.php

```diff
@@ -226,26 +228,51 @@ private function _call( $path, $method = 'GET', $params = array(), $retry = fals
 
                 if ( null !== $result &&
                      isset( $result->error ) &&
-                     isset( $result->error->code ) &&
-                     'request_expired' === $result->error->code
+                     isset( $result->error->code )
                 ) {
-                    if ( ! $retry ) {
-                        $diff = isset( $result->error->timestamp ) ?
-                            ( time() - strtotime( $result->error->timestamp ) ) :
-                            false;
-
-                        // Try to sync clock diff.
-                        if ( false !== $this->_sync_clock_diff( $diff ) ) {
-                            // Retry call with new synced clock.
-                            return $this->_call( $path, $method, $params, true );
+                    $retry = false;
+
+                    if ( ! $in_retry ) {
+                        if ( 'request_expired' === $result->error->code ) {
+                            $diff = isset( $result->error->timestamp ) ?
+                                ( time() - strtotime( $result->error->timestamp ) ) :
+                                false;
+
+                            // Try to sync clock diff.
+                            if ( false !== $this->_sync_clock_diff( $diff ) ) {
+                                // Retry call with new synced clock.
+                                $retry = true;
+                            }
+                        } else if (
+                            Freemius_Api_WordPress::IsHttps() &&
+                            ! empty( $result->error->message ) &&
+                            ( false !== strpos( $result->error->message, 'ssl' ) ) &&
+                            ( false !== strpos( $result->error->message, 'curl error 35' ) )
+                        ) {
+                            $force_http = true;
+                            $retry      = true;
+                        }
+                    }
+
+                    if ( $retry ) {
+                        if ( $force_http ) {
+                            self::$_options->set_option( 'api_force_http', true, true );
                         }
+
+                        $result = $this->_call( $path, $method, $params, true );
                     }
                 }
             }
 
-            if ( $this->_logger->is_on() && self::is_api_error( $result ) ) {
-                // Log API errors.
-                $this->_logger->api_error( $result );
+            if ( self::is_api_error( $result ) ) {
+                if ( $this->_logger->is_on() ) {
+                    // Log API errors.
+                    $this->_logger->api_error( $result );
+                }
+
+                if ( $force_http ) {
+                    self::$_options->set_option( 'api_force_http', false, true );

```

Comment:
@fajardoleo, similarly, I think we need to update `Freemius_Api_WordPress` to revet back to HTTPS. I guess we need to add `Freemius_Api_WordPress::SetHttps()` as a helper method.